Source/Core/InputCommon/ControllerInterface/XInput/XInput.h

```diff
@@ -90,7 +90,7 @@ class Device : public Core::Device
 
 private:
 	XINPUT_STATE m_state_in;
-	XINPUT_VIBRATION m_state_out;
```

Comment:
This can be styled as

```cpp
XINPUT_VIBRATION m_state_out{};
XINPUT_VIBRATION m_current_state_out{};
```

Which would let you get rid of both explicit ZeroMemory calls in the constructor.